homeassistant/components/azure_devops/sensor.py

```diff
@@ -2,89 +2,227 @@
 
-    build_key: int
-    attrs: Callable[[DevOpsBuild], Any]
-    value: Callable[[DevOpsBuild], StateType]
+
+@dataclass(frozen=True, kw_only=True)
+class AzureDevOpsBuildSensorEntityDescription(
+    AzureDevOpsEntityDescription,
+    AzureDevOpsBaseBuildSensorEntityDescription,
+):
+    """Class describing Azure DevOps build sensor entities."""
+
+    item_key: int = 0
+
+
+BASE_BUILD_SENSOR_DESCRIPTIONS: tuple[
+    AzureDevOpsBaseBuildSensorEntityDescription, ...
+] = (
+    AzureDevOpsBaseBuildSensorEntityDescription(
+        key="latest_build",
+        translation_key="latest_build",
+        attrs=lambda build: {
+            "definition_id": (build.definition.build_id if build.definition else None),
+            "definition_name": (build.definition.name if build.definition else None),
+            "id": build.build_id,
+            "reason": build.reason,
+            "result": build.result,
+            "source_branch": build.source_branch,
+            "source_version": build.source_version,
+            "status": build.status,
+            "url": build.links.web if build.links else None,
+            "queue_time": build.queue_time,
+            "start_time": build.start_time,
+            "finish_time": build.finish_time,
+        },
+        value=lambda build: build.build_number,
+    ),
+    AzureDevOpsBaseBuildSensorEntityDescription(
+        key="build_id",
+        translation_key="build_id",
+        entity_registry_visible_default=False,
+        attrs=None,
+        value=lambda build: build.build_id,
+    ),
+    AzureDevOpsBaseBuildSensorEntityDescription(
+        key="reason",
+        translation_key="reason",
+        entity_registry_visible_default=False,
+        attrs=None,
+        value=lambda build: build.reason,
+    ),
+    AzureDevOpsBaseBuildSensorEntityDescription(
+        key="result",
+        translation_key="result",
+        entity_registry_visible_default=False,
+        attrs=None,
+        value=lambda build: build.result,
+    ),
+    AzureDevOpsBaseBuildSensorEntityDescription(
+        key="source_branch",
+        translation_key="source_branch",
+        entity_registry_enabled_default=False,
+        entity_registry_visible_default=False,
+        attrs=None,
+        value=lambda build: build.source_branch,
+    ),
+    AzureDevOpsBaseBuildSensorEntityDescription(
+        key="source_version",
+        translation_key="source_version",
+        entity_registry_visible_default=False,
+        attrs=None,
+        value=lambda build: build.source_version,
+    ),
+    AzureDevOpsBaseBuildSensorEntityDescription(
+        key="status",
+        translation_key="status",
+        entity_registry_visible_default=False,
+        attrs=None,
+        value=lambda build: build.status,
+    ),
+    AzureDevOpsBaseBuildSensorEntityDescription(
+        key="queue_time",
+        translation_key="queue_time",
+        device_class=SensorDeviceClass.DATE,
```

Comment:
Why is this a date?

Reply:
Is there a datetime data class equivilant? os should this technically be converted to a timestamp?

Reply:
I think timestamp is the way to go